ckanext/recombinant/logic.py

```diff
@@ -258,10 +258,28 @@ def _update_datastore(lc, geno, dataset, force_update=False):
 
         trigger_names = _update_triggers(lc, chromo)
 
+        chromo_foreign_keys = chromo.get('datastore_foreign_keys', None)
+        foreign_keys = None
```

Comment:
I think the behaviour we want from datastore_create is that all foreign keys are removed when we pass an empty dict, so here and below we should default to `{}` not `None`